### DISCUSSION TITLE
v4 to v5 migration

### DISCUSSION THREAD

**Discussion by bosticka** [feathersjs/feathers#3082] [General]
I have a v4 feathersjs app with about 20 services or so.  I updated to the v5 packages and made the necessary changes to get it to run.

However, when I ran feathers generate service, it errored out because of some differences between es5 to es6 require/import stuff.

Looking deeper into v5, beyond the change to es6 imports, it has schemas and resolvers.  So migrating my app to this seems unlikely at this point.

I feel like I could create a blank v5 app and then move my models/services/hooks over one by one seems safer and faster.

Has anyone migrated a v4 app to v5?  Seems daunting...

**Comment by daffl** [comment]
Most changes other than the (smaller) breaking changes outlined in [the migration guide](https://feathersjs.com/guides/migrating.html) are backwards compatible. So while the new structure is different, schemas, resolvers and ES6 imports etc. are - as most things in Feathers - optional to use.

You should be able to upgrade the versions as outlined in the migration guide (without changing the import syntax) and continue to be able generate services with the v4 CLI and they should continue to work as is. Once all ecosystem modules are updated there will also be more information on how a further upgrade might look like.
